app/workers/asset_manager_create_whitehall_asset_worker.rb

```diff
@@ -21,6 +21,14 @@ def perform(file_path, legacy_url_path, draft = false, model_class = nil, model_
     end
 
     asset_manager.create_whitehall_asset(asset_options)
+
+    if model
+      Attachment.where(attachable: model.attachables).find_each do |attachment|
+        next unless attachment.attachment_data
```

Comment:
Can we do `Attachment.where(attachable: model.attachables).where.not(attachment_data: nil)` in the query instead? Then we could have a scope that's like `.has_attachment_data`.